widget_preferences.py

```diff
@@ -87,7 +101,11 @@ def export(self, id:str) -> Dict[str, str]:
         dict[id + '_alignment'] = self.alignment
         dict[id + '_expand_direction'] = self.expand_direction
         dict[id + '_minimized'] = "1" if self.minimized else "0"
-        
+
+        for extra_preference in self.extra_preferences:
```

Comment:
This requires an extra hasattr check - Tested this with my current preferences and this caused a crash.

Reply:
Should we instead add a default value to `ExtraPreference` and set that in the constructor the way we do for the other args?

Reply:
Yeah that could work as well, and seems like a good place to put a default